apps/meteor/tests/end-to-end/api/28-roles.ts

```diff
@@ -140,4 +144,279 @@ describe('[Roles]', function () {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('error', 'User does not have the permissions required for this action [error-unauthorized]');
+				});
+		});
+
+		it('should successfully delete a role in EE', async function () {
+			// TODO this is not the right way to do it. We're doing this way for now just because we have separate CI jobs for EE and CE,
+			// ideally we should have a single CI job that adds a license and runs both CE and EE tests.
+			if (!isEnterprise) {
+				this.skip();
+				return;
+			}
+
+			await request
+				.post(api('roles.delete'))
+				.set(credentials)
+				.send({
+					roleId: testRoleId,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res: Response) => {
+					expect(res.body).to.have.property('success', true);
+				});
+		});
+	});
+
+	describe('[/roles.removeUserFromRole]', () => {
+		let testUser: IUser;
+		let testUserCredentials: { 'X-Auth-Token': string; 'X-User-Id': string };
+		const testRoleName = `role.removeUsersFromRole.${Date.now()}`;
+		let testRoleId = '';
+
+		before(async () => {
+			await updatePermission('access-permissions', ['admin']);
+			testUser = await createUser();
+			testUserCredentials = await login(testUser.username, password);
+			if (!isEnterprise) {
+				return;
+			}
+
+			await request
+				.post(api('roles.create'))
+				.set(credentials)
+				.send({
+					name: testRoleName,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res: Response) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('role');
+					expect(res.body.role).to.have.property('name', testRoleName);
+					testRoleId = res.body.role._id;
+				});
+			await request
+				.post(api('roles.addUserToRole'))
+				.set(credentials)
+				.send({
+					roleId: testRoleId,
+					username: testUser.username,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+				});
+		});
+
+		after(async () => {
+			await request.post(api('roles.delete')).set(credentials).send({
+				roleId: testRoleId,
+			});
+			await deleteUser(testUser);
+		});
+
+		it('should fail removing a user from a role when user does NOT have the access-permissions permission', async () => {
+			await request
+				.post(api('roles.removeUserFromRole'))
+				.set(testUserCredentials)
+				.send({
+					roleId: testRoleId,
+					username: testUser.username,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(403)
+				.expect((res: Response) => {
+					expect(res.body).to.have.property('success', false);
+					expect(res.body).to.have.property('error', 'User does not have the permissions required for this action [error-unauthorized]');
+				});
+		});
+
+		it('should successfully remove a user from a role', async function () {
```

Comment:
add cases for:
- removing a non existent user
- removing an existent user from a non existent role
- removing an existent user from a role user doesnt have